service/controller/resource/endpoint/desired.go

```diff
@@ -35,6 +35,11 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 		}
 	}
 
+	statusAnnotation, hasStatusAnnotation := pod.Annotations[key.AnnotationPodNodeStatus]
+	if hasStatusAnnotation && statusAnnotation != "" {
```

Comment:
Only consider the annotation if it's present (in case the annotation controller isn't working for some reason).